pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -228,38 +185,44 @@ pub mod pallet {
 
 	#[pallet::error]
 	pub enum Error<T> {
+		/// The claim is not ready
+		NotReady,
 		/// Current block great than end block
 		InvalidEndBlock,
 		/// Already init a reward
 		AlreadyInitReward,
-		/// User provide wrong signature
-		InvalidSignature,
-		/// User already associated relay account with native account
-		AlreadyAssociated,
 		/// Already paid all reward
 		AlreadyPaid,
-		/// User not associated relay account with native account yet
-		NoAssociatedAccount,
 		/// User not contribute for the crowdloan
 		NotContributedYet,
-		/// When user try to find reward account not their own
-		BadRelayAccount,
 		/// Invalid conversion while calculating payable amount
 		WrongConversionU128ToBalance,
 		/// User cannot receive a reward
 		RewardFailed,
-		/// The amount of reward is lower than the minimum balance
-		ScantyReward,
+		/// The amount of claim below the minimum balance
+		ClaimAmountBelowMinimum,
 	}
 
 	#[pallet::event]
 	#[pallet::generate_deposit(fn deposit_event)]
 	pub enum Event<T: Config> {
-		AssociatedAccount(T::AccountId, T::RelayChainAccountId),
+		RewardPaid(T::AccountId, BalanceOf<T>),
+	}
 
-		UpdateAssociateAccount(T::AccountId, T::RelayChainAccountId),
+	impl<T: Config> Pallet<T> {
+		/// The account ID of the pallet.
+		///
+		/// This actually does computation. If you need to keep using it, then make sure you cache the
+		/// value and only call this once.
+		pub fn account_id() -> T::AccountId {
+			T::PalletId::get().into_account()
+		}
 
-		RewardPaid(T::AccountId, BalanceOf<T>),
+		pub fn pot() -> BalanceOf<T> {
+			T::RewardCurrency::free_balance(&Self::account_id())
+				// Must never be less than 0 but better be safe.
+				.saturating_sub(T::RewardCurrency::minimum_balance())
```

Comment:
why need to subtract the minimum balance?

Reply:
Got it, but better use check_sub instead of saturating_sub:

```
::RewardCurrency::free_balance(&Self::account_id())
        .checked_sub(&T::Currency::minimum_balance()).unwrap_or_else(Zero::zero)
```